content/en/blog/2023/why-histograms/index.md

```diff
@@ -0,0 +1,86 @@
+data point, but in the world of observability data is usually constantly
+exported. For example, an application may export a histogram every minute which
+summarizes a metric for the previous minute. In this way, you can study how the
+distribution of your data changes over time.
+
+# What are histograms for?
+
+There are many uses for histograms, but their power comes from the ability to
+efficiently answer queries about the distribution of your data. These queries
+most commonly come in some form like "what was the median response time in the
+last minute?" These are known as φ-quantiles, and often are abbreviated in a
+shorthand like `p50` for the 50th percentile or 0.5-quantile, also known as the
+median. More generally, the φ-quantile is the observation value that ranks at
+number φ\*N among the N observations.
+
+# Why is that useful?
+
+The most common use-case for histograms in the observability space is defining
+service level objectives (SLOs). One example of such an SLO might be ">=99% of
+all queries should respond in less than 30ms," or "90% of all page loads should
+become interactive within 100ms of first paint."
+
+In the following chart, you can see the `p50`, `p90`, and `p99` response times
+plotted for some requests over some time. From the data, you can see that 50% of
+requests are served in around 20-30ms or less, 90% of requests are served in
+under about 80ms, and 99% of requests are served in under around 90ms. You can
+very quickly see that at least 50% of your users are receiving very fast
+response times, but almost all of your users are experiencing response times
+under 90ms.
+
+![p99, p90, and p50 plotted as lines](hist-lines.png "p99, p90, and p50 plotted as a line chart with title 'response times.' Time is on the x-axis and response times in milliseconds on the y-axis. p99 response times are around 80 milliseconds. p90 response times are betweeen 60 and 80 milliseconds. p50 response times are between 20 and 30 milliseconds.")
+
+# Other metric types
+
+Another solution might be to define the SLOs you're interested in and collect
+them as non-histogram metrics in advance as gauges or counters. This approach
+works, but requires defining your SLOs before you have an understanding of your
+data distribution and requires non-trivial implementation at collection time. It
+is also inflexible; if you decide to change your SLO from 90% of requests to 99%
+of requests, you have to make and release code changes, then wait for the old
+data to age out and the new metric to collect enough data to make useful
+queries. Because histograms model data as a distribution from start to finish,
+they enable you to simply change your queries and get answers on the data you've
+already collected. Particularly with exponential histograms, arbitrary
+distribution queries can be made with very low relative error rates and minimal
+resource consumption on both the client and the analysis backend.
```

Comment:
```suggestion
What if you're already defining SLOs based on other metrics?
You may have considered defining the SLOs to be based on gauges or counters.
This approach can work, but it requires defining your SLOs before understand your
data distribution and requires non-trivial implementation at collection time. It
is also inflexible; if you decide to change your SLO from 90% of requests to 99%
of requests, you have to make and release code changes, then wait for the old
data to age out and the new metric to collect enough data to make useful
queries. Because histograms model data as a distribution from start to finish,
they enable you to simply change your queries and get answers on the data you've
already collected. Particularly with exponential histograms, arbitrary
distribution queries can be made with very low relative error rates and minimal
resource consumption on both the client and the analysis backend.
```